Samples~/SampleGame/Assets/Editor/Scripts/ClientServerSwitchMenu.cs

```diff
@@ -95,6 +96,7 @@ private static void Switch(Action updateDefines)
 
         EditorSceneManager.OpenScene(BootstrapScenePath);
         updateDefines();
+        CompilationPipeline.RequestScriptCompilation();
```

Comment:
I guess I'm not following how this fixes the bug you linked this to in the tracker. The bug was the defines were not being updated in the player settings, so wouldn't compiling the scripts still compile with the current/previous defines?

Reply:
We noted that there is occasionally an issue where you will select these menu items and the define will change but the compilation engine will not update, this forces that.

Reply:
To be clear - this updates the defines when you explicitly choose the menu item. However, if you did not choose a menu item, the player settings itself are incorrect. 

From our discussion it seemed like that should be populated automatically without having to explicitly choose the menu item if possible.

Reply:
I tried testing this locally and am seeing more discrepancies:

1. "Apply Windows Server Settings" sets the `UNITY_SERVER` flag for Win/Mac/Linux (which I'm guessing is the client) and clears it for Dedicated Server
2. "Apply Client Server Settings" resets it for both the options

Reply:
I have commited to using the build sub target within unity. The defines are set once using the menu item (that i just added to the server sub target. When you swap build target between client and server now it will automatically update the subtarget and use the other define